### ISSUE TITLE
Re:HDF5Dataset

### ISSUE THREAD

**Issue by mdhe1248** [HolyLab/RegisterDeformation.jl#13]
Hi all, 
I hope all is well.

I tried RegisterQD and got an error message during `using RegisterQD`:
```jl
julia> using RegisterQD
[ Info: Precompiling RegisterQD [ac24ea0c-1830-11e9-18d4-81f172323054]
WARNING: could not import CoordinateTransformations.transform into RegisterDeformation
ERROR: LoadError: UndefVarError: HDF5Dataset not defined
Stacktrace:
 [1] top-level scope at /home/donghoon/.julia/packages/RegisterDeformation/IWvZQ/src/RegisterDeformation.jl:587
 [2] include(::Function, ::Module, ::String) at ./Base.jl:380
 [3] include(::Module, ::String) at ./Base.jl:368
 [4] top-level scope at none:2
 [5] eval at ./boot.jl:331 [inlined]
 [6] eval(::Expr) at ./client.jl:467
 [7] top-level scope at ./none:3
in expression starting at /home/donghoon/.julia/packages/RegisterDeformation/IWvZQ/src/RegisterDeformation.jl:587
ERROR: LoadError: Failed to precompile RegisterDeformation [c19381b7-cf49-59d7-881c-50dfbd227eaf] to /home/donghoon/.julia/compiled/v1.5/RegisterDeformation/PU4nJ_PlBN4.ji.
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] compilecache(::Base.PkgId, ::String) at ./loading.jl:1305
 [3] _require(::Base.PkgId) at ./loading.jl:1030
 [4] require(::Base.PkgId) at ./loading.jl:928
 [5] require(::Module, ::Symbol) at ./loading.jl:923
 [6] include(::Function, ::Module, ::String) at ./Base.jl:380
 [7] include(::Module, ::String) at ./Base.jl:368
 [8] top-level scope at none:2
 [9] eval at ./boot.jl:331 [inlined]
 [10] eval(::Expr) at ./client.jl:467
 [11] top-level scope at ./none:3
in expression starting at /home/donghoon/.julia/packages/RegisterQD/ZBH17/src/RegisterQD.jl:6
ERROR: Failed to precompile RegisterQD [ac24ea0c-1830-11e9-18d4-81f172323054] to /home/donghoon/.julia/compiled/v1.5/RegisterQD/FExqP_PlBN4.ji.
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] compilecache(::Base.PkgId, ::String) at ./loading.jl:1305
 [3] _require(::Base.PkgId) at ./loading.jl:1030
 [4] require(::Base.PkgId) at ./loading.jl:928
 [5] require(::Module, ::Symbol) at ./loading.jl:923

```
I tracked the error message, and it seems that `HDF5Dataset` should be `HDF5.Dataset` in the `warp!` function. Please correct me if I am wrong.

It is a simple fix, I waned to submit a pull request, but I was not sure how to do it. The repository management has been so advance that I need to learn it. If anyone has some spare time, please guide me.

**Comment by kdw503** [comment]
Hi @mdhe1248.
Thank you for the simple fix tip.
It looks HDF5Dataset is changed HDF5.Dataset after HDF5@0.14.
So, I think we need to change the HDF5 compat also in the Project.toml file

**Comment by timholy** [comment]
Thanks for reporting this!

This is us being hit-and-miss about upgrading and/or specifying dependencies. Here's the brief summary:
- https://github.com/HolyLab/RegisterDeformation.jl/blob/c5c4f54beb8518621dd46b8568f5365a64e875c2/Project.toml#L34 means that it should not choose HDF5 0.14, because that hasn't been declared compatible. Thus it's quite a puzzling bug initially.
- However, version compatibility bounds got added only in https://github.com/HolyLab/RegisterDeformation.jl/pull/4. That was released in v0.3.0 of RegisterDeformation.
- [RegisterQD still requires v0.2 of RegisterDeformation](https://github.com/HolyLab/RegisterQD.jl/blob/c8511747dd2fa5306d858739c1ff97987d44f4b5/Project.toml#L31) :frowning: 

Consequently, there's nothing preventing it from installing an incompatible version of HDF5.

While #14 is a very good change (and we should merge that), it won't actually fix this problem. That's because users of RegisterQD will still be stuck on RegisterDeformation 0.2, and won't get the fix.  https://github.com/HolyLab/RegisterQD.jl/pull/29 should fix it, though, once we tag a new release (which I'll do as soon as that gets through testing).

**Comment by timholy** [comment]
OK, now you should be able to `pkg>up` and everything should be fixed.
